src/Emulator/Peripherals/Peripherals/UART/STM32_UART.cs

```diff
@@ -30,15 +30,28 @@ public void WriteChar(byte value)
                 this.Log(LogLevel.Warning, "Received a character, but the receiver is not enabled, dropping.");
                 return;
             }
-            receiveFifo.Enqueue(value);
-            readFifoNotEmpty.Value = true;
-            Update();
+            lock (receiveFifo)
+            {
+                receiveFifo.Enqueue(value);
+                readFifoNotEmpty = true;
+
+                machine.ClockSource.ExecuteInLock(() =>
+                {
+                    if (receiverNotEmptyInterruptEnabled.Value)
```

Comment:
FYI:

So there is the case, that the machine is not yet done with the last interrupt handling:

_receiverNotEmptyInterruptEnabled_ is not yet set to true but _transmissionCompleteInterruptEnabled_ and _transmissionComplete_ are so the Update() method would wrongly set IRQ to true again.

If this happens, the IRQ hangs somehow.

Overall I am not a fan of updating the IRQ from a different thread than the machine thread.